server/src/main/java/io/atomix/copycat/server/state/AbstractAppender.java

```diff
@@ -322,9 +332,11 @@ protected void updateMatchIndex(MemberState member, AppendResponse response) {
   /**
    * Updates the next index when the match index is updated.
    */
-  protected void updateNextIndex(MemberState member) {
+  protected void updateNextIndex(MemberState member, AppendRequest request) {
     // If the match index was set, update the next index to be greater than the match index if necessary.
-    member.setNextIndex(Math.max(member.getMatchIndex() + 1, 1));
+    if (!request.entries().isEmpty()) {
+      member.setNextIndex(request.entries().get(request.entries().size()-1).getIndex()+1);
```

Comment:
Update `nextIndex` based on the last appended entry's index + 1. We have to use the last entry rather than the entry count since incremental compaction may exclude entries from replication.